enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,210 @@
+---
+title: upgrade-acknowledgment-gate
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@bparees"
+  - "@sdodson"
+  - "@csrwng"
+  - "@relyt0925"
+  - "@jeremyeder"
+approvers:
+  - TBD 
+creation-date: 2021-07-08
+last-updated: 2021-07-14
+status: implementable
+---
+
+# Upgrade Acknowledgment Gate
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently blocks minor level upgrades when overrides are set, a resource deletion is pending, or an operator has set its upgradeable condition to false.
+This enhancement adds a generic upgrade blocking mechanism for blocking minor level upgrades.
+This generic upgrade blocker can consist of one or more items each of which will block a minor level upgrade.
+Each generic upgrade blocker item will be associated with some well-defined task to be completed by an administrator.
+Once the task has been completed the administrator will acknowledge completion and its associated item will no longer block minor level upgrades.
+
+## Motivation
+
+OCP 4.9 uses kubernetes v1.22 which is removing a large number of [APIs][k8s-1.22-removed-apis].
+We suspect that many customer clusters are still using these APIs through helm charts, operators, generic workloads, administrative tools, etc.
+We believe there is no effective way for a Cluster Operator to detect the use of these APIs by external applications.
+Relying solely on automated API detection to block upgrades could result in blocking for APIs that are not actively used any longer or not blocking because an application was not active during the check.
+Having a human in the loop to drive the analysis and make the final decision is the surest way to prevent customers from upgrading from 4.8.z to 4.9.z and encountering missing API related breakage.
```

Comment:
This doesn't prevent upgrades when API related breakage will happen, it moves the responsibility to the administrator.

Reply:
correct....are you questioning that as an axiom, or looking for better phrasing, or something else?

fundamentally this EP represents us saying "we cannot solve this problem for admins, therefore we need admins to be accountable for addressing it, with our best guidance"

Reply:
Better phrasing, shifting responsibility sounds like the best approach in this situation.

```suggestion
Having a human in the loop to drive the analysis and make the final decision is the surest way to prevent customers from upgrading from 4.8.z to 4.9.z and reduce instances of missing API related breakage by allowing administrators to assess the impact per cluster and impacted application.
```

Reply:
Reworded with slight change. Our goal here is to eliminate API breakage rather than to just reduce.